docs/android/v2/how-to-guides/extend-capabilities/plugins/blur-filter.mdx

```diff
@@ -0,0 +1,120 @@
+---
+title: Blur Filter (Beta)
+nav: 14.3
+---
+
+import AndroidSdkVersionShield from '@/common/android-sdk-version-shield.md';
+
+## Introduction
+
+Blur filter plugin helps in bluring the non-face background of a video frame.
+This guide provides an overview of usage of the Blur Filter plugin of 100ms.
+
+
+## Supported Versions/Resolutions
+
+-   Minimum 100ms SDK version it can work with is `2.9.2`
+-   Works best on 15fps
+-   Has poor fps on older android phones
+
+
+## Add dependency
+
+-   Adding the Blur plugin and SDK dependency to your app-level `build.gradle`.
+
+<AndroidSdkVersionShield />
+
+<Tabs id="sdk-imports" items={['Maven Central']} />
+
+<Tab id='sdk-imports-0'>
+
+```json{5-8}:build.gradle section=AddSDKDependencies sectionIndex=1
+dependencies {
+ // See the version in the badge above.
+
+ def hmsVersion = "x.x.x"
+ implementation "live.100ms:android-sdk:$hmsVersion"    // Essential
+ implementation "live.100ms:virtual-background:$hmsVersion" // Optional
+}
+```
+
+</Tab>
+
+
+## How to Integrate Blur Filters
+
+### Instantiate 
+
+Instantiate the 100ms Blur Filter plugin like this:
+
+<Tabs id="instanstiate-VirtualBackgroundPlugin" items={['Kotlin']} />
+
+<Tab id='instanstiate-VirtualBackgroundPlugin-0'>
+
+```kotlin
+     val hmsSDK = HMSSDK
+        .Builder(application)
+        .build()
+
+     val blurFilterPlugin  by lazy { HMSBlurFilter(hmsSDK) }
+
+     //call this after onJoin()
+     fun addBeautyFilterPlugin() {
+
+        if (hmsSDK.getLocalPeer()?.videoTrack != null) {
+
+            hmsSDK.addPlugin(blurFilterPlugin, object : HMSActionResultListener {
```

Comment:
Do add that "THis is a async operation. It takes a few seconds on some devices to start. So UI should add a spinner indicating that adding of blur is in progress"